File: lib/lunaconf/log.lua
-- Logging via the log functions here will log to syslog.
-- If you use systemd you can easily follow the log with:
-- $ journalctl -t awesome -f

local awful = require('awful')

local log = {}

function log.log(level, message, ...)
	awful.spawn.spawn('logger -t awesome "' .. string.format(message, ...) .. '"')
end

function log.info(message, ...)
	log.log('info', message, ...)
end

function log.notice(message, ...)
	log.log('notice', message, ...)
end

function log.warning(message, ...)
	log.log('warning', message, ...)
end

function log.err(message, ...)
	log.log('err', message, ...)
end

return log

File: src/lib/log.ts
import * as awful from 'awful';

const LOGGER_TAG = 'awesome';

export enum LogLevel {
  EMERG = 0,
  ALERT = 1,
  CRIT = 2, 
  ERR = 3,
  WARNING = 4,
  NOTICE = 5,
  INFO = 6,
  DEBUG = 7,
};

export function log(msg: string, level: LogLevel = LogLevel.INFO) {
  awful.spawn.spawn(`logger -t ${LOGGER_TAG} -p ${level} "${msg.replaceAll('"', '\\"')}"`);
}